elastic/kibana
# [Stack Monitoring] Enable OOTB alerts in RAC page and multiple rules of a rule type

### Summary

Resolves https://github.com/elastic/kibana/issues/91145
Resolves https://github.com/elastic/kibana/issues/100142
Resolves https://github.com/elastic/kibana/issues/92128

- Enables OOTB alerts to be managed in Stack Management -> RAC
- Adds support for allowing multiples rules of a rule type

### Test
- Test alert menus with 2 rules of the same type
  - Create the OOTB default alerts first when prompted by the modal or use the dropdown 
  - Edit one of the rules like CPU Usage to have something very low to trigger an alert.  Select "notify me when alert is active" and to notify very 1 minute. Make sure an alert gets created.
  - Navigate to Stack Management -> Rules and Connectors
  - click "Create Rule".  In the Flytout name the new rule (eg CPU USage 2) and select the same rule modified earlier like CPU Usage under Stack Monitoring and use the same settings as above.  Wait for alerts to be created.
  - Click on orange "x alerts" badge.  Make sure its set to "Group by Alert Type"
  - You should now see in the menu both rules of the same rule type listed, with each node with the alert nested under it
  - ![Screen Shot 2021-07-21 at 3 49 45 PM](https://user-images.githubusercontent.com/1676003/126551072-022b777f-921a-4edb-9415-44b310e1b9df.png)
  - Click on orange "x alerts" badge.  Make sure its set to "Group by Node Type"
  - You should see a list of each node with the number of alerts it has in parenthesis 
  - Clicking on the node you should see each rule that's being alerted for that node
  - ![Screen Shot 2021-07-21 at 3 53 34 PM](https://user-images.githubusercontent.com/1676003/126551534-4c091c97-567a-49fb-999d-4a7215ee830a.png)
  - Assure that the count "x alerts" on the badge, matches the actual alerts in the menu
- Test other pages
  -  Clicking on any of the pages such as Nodes you should not have any errors.  The alerts should be in a callout at the top and the alert count should match the number of alerts
  - ![Screen Shot 2021-07-21 at 3 55 22 PM](https://user-images.githubusercontent.com/1676003/126551830-8b8fbb5e-d926-45e7-bf20-46d19314378d.png)
  - The detailed Node page has alerts for all the rules (note 2 CPU usage alerts) 
![Screen Shot 2021-07-22 at 12 40 39 PM](https://user-images.githubusercontent.com/1676003/126676376-9195f00f-bfb7-4a6f-b0ed-9faea8c79d7b.png)
